crates/oxc_linter/src/rules/eslint/no_constructor_return.rs

```diff
@@ -0,0 +1,123 @@
+use oxc_ast::{
+    ast::{MethodDefinition, MethodDefinitionKind},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::AstNodeId;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_constructor_return_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(
+        "eslint(no-constructor-return): Unexpected return statement in constructor.",
+    )
+    .with_labels([span.into()])
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoConstructorReturn;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow returning value from constructor
+    ///
+    /// ### Why is this bad?
+    ///
+    /// In JavaScript, returning a value in the constructor of a class may be a mistake.
+    /// Forbidding this pattern prevents mistakes resulting from unfamiliarity with the language or a copy-paste error.
+    ///
+    /// ### Example
+    /// Bad:
+    /// ```rust
+    /// class C {
+    ///     constructor() { return 42; }
+    /// }
+    /// ```
+    ///
+    /// Good:
+    /// ```rust
+    /// class C {
+    ///     constructor() { this.value = 42; }
+    /// }
+    /// ```
+    NoConstructorReturn,
+    correctness
+);
+
+impl Rule for NoConstructorReturn {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::ReturnStatement(ret) = node.kind() else { return };
+        if ret.argument.is_none() {
+            return;
+        }
+
+        let Some(is_in_constructor_root) = is_in_constructor_root(ctx, node.id()) else { return };
+
+        if is_in_constructor_root || is_definitely_in_constructor(ctx, node.id()) {
```

Comment:
I think we don't need the `is_in_constructor_root` check anymore; So it can be removed along with its function, Other than that it looks good to me.

Thanks again for this contribution and your patience.

Reply:
ok, got it.